cluster.yml

```diff
@@ -38,7 +38,11 @@
   roles:
     - { role: kubespray-defaults }
     - { role: kubernetes/preinstall, tags: preinstall }
-    - { role: "container-engine", tags: "container-engine", when: deploy_container_engine|default(true) }
+    - role: "container-engine"
+      tags: "container-engine"
+      when:
+        - deploy_container_engine|default(true)
+        - etcd_deployment_type != "host" or "k8s-cluster" in group_names
```

Comment:
I think you should define `deploy_container_engine` in `roles/kubespray-defaults/defaults/main.yaml`, because `deploy_container_engine` is used at multiple locations
something like
```
deploy_container_engine: etcd_deployment_type != "host" or inventory_hostname in groups['kube-cluster']
```